src/core/electric_utility.jl

```diff
@@ -29,35 +29,56 @@
 # *********************************************************************************
 """
     ElectricUtility
-
 ElectricUtility data struct with inner constructor:     
+        
 ```julia
 function ElectricUtility(;
-    outage_start_time_step::Int=0,  # for modeling a single outage, with critical load spliced into the baseline load ...
-    outage_end_time_step::Int=0,  # ... utiltity production_factor = 0 during the outage
-    allow_simultaneous_export_import::Bool = true,  # if true the site has two meters (in effect)
-    # variables below used for minimax the expected outage cost,
+    emissions_region::String = "",
+    emissions_factor_series_lb_CO2_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+    emissions_factor_series_lb_NOx_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+    emissions_factor_series_lb_SO2_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+    emissions_factor_series_lb_PM25_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+    emissions_factor_CO2_decrease_pct::Float64 = 0.01174,
+    emissions_factor_NOx_decrease_pct::Float64 = 0.01174,
+    emissions_factor_SO2_decrease_pct::Float64 = 0.01174,
+    emissions_factor_PM25_decrease_pct::Float64 = 0.01174,
+    outage_start_time_step::Int=0  # for modeling a single outage, with critical load spliced into the baseline load ...
+    outage_end_time_step::Int=0  # ... utiltity production_factor = 0 during the outage
+    allow_simultaneous_export_import::Bool=true  # if true the site has two meters (in effect)
+    # next 5 variables below used for minimax the expected outage cost,
     # with max taken over outage start time, expectation taken over outage duration
     outage_start_time_steps::Array{Int,1}=Int[],  # we minimize the maximum outage cost over outage start times
     outage_durations::Array{Int,1}=Int[],  # one-to-one with outage_probabilities, outage_durations can be a random variable
     outage_probabilities::Array{R,1} where R<:Real = [1.0],
     outage_time_steps::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:maximum(outage_durations),
     scenarios::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:length(outage_durations),
     net_metering_limit_kw::Real = 0,
-    interconnection_limit_kw::Real = 1.0e9
-)
-```julia
-
+    interconnection_limit_kw::Real = 1.0e9,
+    latitude::Real,
+    longitude::Real,
+    time_steps_per_hour::Int = 1
+    )
+```
 !!! note Outage indexing begins at 1 (not 0) and the outage is inclusive of the outage end time step. 
     For instance, to model a 3-hour outage from 12AM to 3AM on Jan 1, outage_start_time_step = 1 and outage_end_time_step = 3.
     To model a 1-hour outage from 6AM to 7AM on Jan 1, outage_start_time_step = 7 and outage_end_time_step = 7.
 
 """
- struct ElectricUtility
+mutable struct ElectricUtility
+    emissions_region::String
+    distance_to_emissions_region_meters::Real
```

Comment:
the value for `distance_to_emissions_region_meters` can be `nothing` (returned from the `region_abbreviation` function), which would break here and in the results/electric_utility.jl. Should there be a default high number for the distance? We could return it from the `region_abbreviation` function.

Reply:
I changed the field type to accept nothing

Reply:
Oh I see, that's what you meant by breaking results/electric_utility.jl. In commit afd2191 I've made distance_to_emissions_region_meters = typemax(Int64) instead of nothing.